clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -1028,6 +1045,7 @@ void checkExprLifetime(Sema &SemaRef, const InitializedEntity &Entity,
 
       switch (shouldLifetimeExtendThroughPath(Path)) {
       case PathLifetimeKind::Extend:
+        assert(InitEntity && "Expect only on initializing the entity");
```

Comment:
Maybe make it clearer that we do not expect lifetime extension for assignment:
 "Lifetime extension should happen only for initialization and not assigment". Same below.